hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/OperationRunner.java

```diff
@@ -14,38 +14,37 @@
  * limitations under the License.
  */
 
-package com.hazelcast.spi.impl;
+package com.hazelcast.spi.impl.operationexecutor;
 
 import com.hazelcast.nio.Packet;
 import com.hazelcast.spi.Operation;
 
 /**
- * The OperationHandler is responsible for the actual processing of operations.
+ * The OperationRunner is responsible for the actual running of operations.
  * <p/>
- * So the {@link com.hazelcast.spi.impl.OperationScheduler} is responsible for scheduling them (so finding a thread
- * to run on), the actual work is done by the {@link com.hazelcast.spi.impl.OperationHandler}.
+ * So the {@link OperationExecutor} is responsible for 'executing' them (so finding a thread to run on), the actual work is done
+ * by the {@link OperationRunner}. This separation of concerns makes the code a lot simpler and makes it possible to swap parts
+ * of the system.
  * <p/>
- * Since HZ 3.5 there are multiple OperationHandler instances; each partition will have its own OperationHandler, but also
- * generic threads will have their own OperationHandlers. Each OperationHandler exposes the Operation it is currently working
+ * Since HZ 3.5 there are multiple OperationRunner instances; each partition will have its own OperationRunner, but also
+ * generic threads will have their own OperationRunners. Each OperationRunner exposes the Operation it is currently working
  * on and this makes it possible to hook on all kinds of additional functionality like detecting slow operations, sampling which
  * operations are executed most frequently, check if an operation is still running, etc etc.
  *
  * Why is this class an abstract class and not an interface?
  *
- * If this class would be an interface, the {@link #currentTask()} will be more expensive. THis method is going to be called
- * 'frequently'; if there are 5000 partitions, then there will be 5000 calls to this method. Since it is likely that there are
- * multiple implementations of this OperationHandler, there will be virtual method call that can't be inlined if this class
- * would be an interface.
+ * If this class would be an interface, the {@link #currentTask()} will be more expensive. This method is going to be called
```

Comment:
Not everything needs to be backed up by benchmarks. Common sense is good enough.

In this case it is fine to have an abstract class. There is no added value of having an interface with the risk of running into megamorphic method calls. With the current approach it can't happen.